static/js/components/source/source_publish/SourcePublishHistory.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState } from "react";
+import makeStyles from "@mui/styles/makeStyles";
+import { useDispatch } from "react-redux";
+import PropTypes from "prop-types";
+import CircularProgress from "@mui/material/CircularProgress";
+import DeleteIcon from "@mui/icons-material/Delete";
+import Link from "@mui/material/Link";
+import moment from "moment";
+import {
+  deletePublicSourcePage,
+  fetchPublicSourcePages,
+} from "../../../ducks/public_pages/public_source_page";
+
+const useStyles = makeStyles(() => ({
+  versionHistory: {
+    width: "100%",
+  },
+  versionHistoryLine: {
+    border: "1px solid #e0e0e0",
+    display: "flex",
+    padding: "0.25rem 0.5rem",
+    justifyContent: "space-between",
+    alignItems: "center",
+    borderRadius: "0.5rem",
+    marginBottom: "0.5rem",
+  },
+  noVersion: {
+    display: "flex",
+    justifyContent: "center",
+    padding: "1rem 0",
+    color: "gray",
+    fontWeight: "bold",
+  },
+}));
+
+const SourcePublishHistory = ({ source_id, newPageGenerate = null }) => {
+  const styles = useStyles();
+  const dispatch = useDispatch();
+  const [versions, setVersions] = useState([]);
+  const [isVersions, setIsVersions] = useState(true);
+
+  useEffect(() => {
+    dispatch(fetchPublicSourcePages(source_id, 10)).then((data) => {
+      setVersions(data.data);
+      setIsVersions(data.data.length > 0);
+    });
+  }, [dispatch, source_id]);
+
+  useEffect(() => {
+    if (newPageGenerate) {
+      setVersions([{ PublicSourcePage: newPageGenerate }, ...versions]);
+      setIsVersions(true);
+    }
+  }, [newPageGenerate]);
+
+  const deleteVersion = (id) => {
+    dispatch(deletePublicSourcePage(id)).then((data) => {
+      if (data.status === "success") {
+        setIsVersions(versions.length > 1);
+        setVersions(versions.filter((obj) => obj.PublicSourcePage.id !== id));
+      }
+    });
+  };
+  const displayDate = (date) => {
+    // Parse the date with Moment.js
+    const dateObj = moment(date);
+
+    // Format the date into a string with up to 2 fractional second digits
+    const dateString = dateObj.format("MM/DD/YYYY HH:mm:ss");
+
+    // Append ' UTC' to the end of the string
+    return `${dateString} UTC`;
```

Comment:
the code speaks for itself here, you can remove the comment

Reply:
✅ Done.